cli/golem-cli/src/app/build/command.rs

```diff
@@ -575,6 +575,19 @@ pub async fn ensure_common_deps_for_tool(
             })
             .await
         }
+        // TODO: this is intentionally not "cargo", rather abusing "ensure once" to do a "prebuild" for now;
```

Comment:
I don't understand this comment. Why are we doing an initial `cargo build` here? And if to make sure agent extraction works, then how is it going to rebuild the agents when needed?

Reply:
@noise64 I guess you could answer this better. I think this is the last commit